src-app/core-modules/app-theme/index.ts

```diff
@@ -1,18 +1,49 @@
+import { useMemo, useReducer } from 'react'
+
+interface State {
+	theme: Theme
+}
+
 const THEMES: Theme[] = ['light', 'dark']
+const DEFAULTS: State = {
+	theme: 'light',
+}
 
-export const loadAppTheme = () => {
-	let theme = localStorage.getItem('theme') as Theme | null
+const reducer = (state: State, action: Partial<State>): State => {
+	const { theme } = action
 
-	if (!theme || !THEMES.includes(theme)) {
-		theme = 'light'
-	}
+	if (theme) return { ...state, theme }
 
-	setAppTheme(theme)
+	return state
 }
 
-export const setAppTheme = (theme: Theme) => {
-	document.documentElement.setAttribute('data-theme', theme)
-	localStorage.setItem('theme', theme)
-}
+export const useAppTheme = () => {
+	const [state, dispatch] = useReducer(reducer, DEFAULTS)
```

Comment:
Use `useContext` instead of `useReducer` - https://react.dev/learn/passing-data-deeply-with-context